src/test/java/ai/rev/speechtotext/integration/CustomVocabulariesTest.java

```diff
@@ -0,0 +1,102 @@
+package ai.rev.speechtotext.integration;
+
+import ai.rev.speechtotext.CustomVocabulariesClient;
+import ai.rev.speechtotext.models.vocabulary.CustomVocabulary;
+import ai.rev.speechtotext.models.vocabulary.CustomVocabularyInformation;
+import ai.rev.speechtotext.models.vocabulary.CustomVocabularyOptions;
+import ai.rev.speechtotext.models.vocabulary.CustomVocabularyStatus;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CustomVocabulariesTest {
+
+  @Rule public TestName testName = new TestName();
+
+  private static final List<String> PHRASES = Arrays.asList("test", "this", "vocab");
+  private CustomVocabulariesClient customVocabularyClient =
+      new CustomVocabulariesClient(EnvHelper.getToken());
+
+  @Test
+  public void
+      submitCustomVocabulary_MetadataAndVocabAreIncluded_ReturnCustomVocabularyInformationInProgress() {
+    CustomVocabularyInformation customVocabularyInformation = submitCustomVocabulary();
+
+    assertThat(customVocabularyInformation.getStatus())
+        .as("Custom vocabulary status")
+        .isEqualTo(CustomVocabularyStatus.IN_PROGRESS);
+  }
```

Comment:
Why is callback not included in this test? I would add it here.

Also I would separate the `customVocabularyInformation.getStatus()` out of the assert statement, please do the same other tests that do this as well. It is useful to have 3 visual blocks in tests:

```
     // Arrange
     CustomVocabularyInformation customVocabularyInformation = submitCustomVocabulary(); 

     // Act
     status = customVocabularyInformation.getStatus()

    // Assert ...
```

Reply:
The callback wasn't included because the integration test really sends a request and I don't have a valid callback url to pass in. This same logic applied to the `SubmitJobTest` integration tests. Should we just add something like `https://test.test.test`?

Are you sure about the blocks? I feel like this is a style preference and adds some unnecessary assignment. I feel like because the everything is named well it's easy to understand and reads well. `Assert that custom vocabulary information status is equal to the custom vocabulary status in_progress`, no?

Also blocks like:
```
for (CustomVocabularyInformation customVocabulary : customVocabularies) {
      assertThat(customVocabulary.getId()).as("Vocabulary Id").isNotNull();
      assertThat(customVocabulary.getCreatedOn()).as("Created on").isNotNull();
      assertThat(customVocabulary.getStatus()).as("Status").isNotNull();
      if (customVocabulary.getStatus() == CustomVocabularyStatus.FAILED) {
        assertThat(customVocabulary.getFailure()).as("Failure").isNotNull();
        assertThat(customVocabulary.getFailureDetail()).as("Failure detail").isNotNull();
      } else if (customVocabulary.getStatus() == CustomVocabularyStatus.COMPLETE) {
        assertThat(customVocabulary.getCompletedOn()).as("Completed on").isNotNull();
      }
    }
```

Will become
```
for (CustomVocabularyInformation customVocabulary : customVocabularies) {
    String id = customVocabulary.getId();
    String createdOn = customVocabulary.getCreatedOn();
    String completedOn = customVocabulary.getCompletedOn();
    CustomVocabularyStatus status = customVocabulary.getStatus();
    CustomVocabularyFailureType failure = customVocabulary.getFailure();
    String failureDetail = customVocabulary.getFailureDetail();

    assertThat(id).as("Vocabulary Id").isNotNull();
    assertThat(createdOn).as("Created on").isNotNull();
    assertThat(status).as("Status").isNotNull();
    if (status == CustomVocabularyStatus.FAILED) {
      assertThat(failure).as("Failure").isNotNull();
      assertThat(failureDetail).as("Failure detail").isNotNull();
    } else if (status == CustomVocabularyStatus.COMPLETE) {
      assertThat(completedOn).as("Completed on").isNotNull();
    }
  }
```

In these two cases I'm not really seeing a big difference but if you think this improves readability I'll make the changes everywhere. 

Will we also need to do it in other SDKs? What about tests in node like https://github.com/revdotcom/revai-node-sdk/blob/develop/test/integration/test/custom_vocabularies.test.js

Reply:
This request was clarified over a zoom and changes have been implemented.